src/lib/Preview.js

```diff
@@ -1269,6 +1310,95 @@ class Preview extends EventEmitter {
         this.viewer.load(err);
     }
 
+    /**
+     * Create a generic log Object.
+     *
+     * @private
+     * @return {Object} Log details for viewer session and current file.
+     */
+    createLog() {
+        const file = this.file || {};
+        const log = {
+            timestamp: getISOTime(),
+            file_id: getProp(file, 'id', ''),
+            file_version_id: getProp(file, 'file_version.id', ''),
+            content_type: getProp(this.viewer, 'options.viewer.NAME', ''),
+            extension: file.extension || '',
+            locale: getProp(this.location, 'locale', ''),
+            ...getClientLogDetails()
+        };
+
+        return log;
+    }
+
+    /**
+     * Message, to any listeners of Preview, that an error has occurred.
+     *
+     * @private
+     * @param {Error} error - The error that occurred.
+     * @return {void}
+     */
+    logPreviewError(error) {
```

Comment:
Maybe different names to differentiate from `logPreviewEvent()` since this isn't actually logging anything while the other function is actually making a log call?

Reply:
I'll rename both to `emitPreviewError()` and `emitLoadMetric()`